sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -735,7 +747,7 @@ class Analyzer(
     }
 
     def lookupTempView(identifier: Seq[String]): Option[LogicalPlan] =
-      identifier match {
+      expandRelationName(identifier) match {
```

Comment:
Do we need expand the relation name here? `lookupTempView` is used to look up temp/global temp view. In permanent views we cannot have temp views.

Reply:
This is to make sure we don't resolve to temp view mistakenly. Imagine that a view refers to a table `t`, and there happens to have a temp view `t`, we shouldn't resolve `t` to temp view when analyzing this view.